File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.java
class Solution {
    public int closetTarget(String[] words, String target, int startIndex) {
        int forward = 0;
        int backward = 0;
        int n = words.length;
        int i = startIndex;
        
        if(!Arrays.asList(words).contains(target))
            return -1;
        
        while(!words[i].equals(target)){
            forward++;
            i=(i+1)%n;
        }
        
        i=startIndex;
        
        while(!words[i].equals(target)){
            backward++;
            i=(i-1+n)%n;
        }
        
        return Math.min(forward,backward);
    }
}

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.py
class Solution(object):
    def closestTarget(self, words, target, startIndex):
        """
        :type words: List[str]
        :type target: str
        :type startIndex: int
        :rtype: int
        """
        if words[startIndex]==target:
            return 0
        
        if target not in words:
            return -1

        n = len(words)
        
        right = 1
        next = (startIndex + 1) % n
        while(words[next]!=target):
            right+=1
            next = (next + 1) % n
        
        left = 1
        next = (startIndex - 1) % n
        while(words[next]!=target):
            left+=1
            next = (next - 1) % n
        
        return min(right,left)